docs/book/src/developer/providers/contracts/infra-machinepool.md

```diff
@@ -0,0 +1,525 @@
+the CRD definition of the InfraMachinePool resource.
+
+It is a generally applied convention to use names in the format `${env}MachinePool`, where ${env} is a, possibly short, name
+for the environment in question. For example `AWSMachinePool` is an implementation for Amazon Web Services, and `AzureMachinePool`
+is one for Azure.
+
+```go
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:path=foomachinepools,shortName=foomp,scope=Namespaced,categories=cluster-api
+// +kubebuilder:storageversion
+// +kubebuilder:subresource:status
+
+// FooMachinePool is the Schema for foomachinepools.
+type FooMachinePool struct {
+    metav1.TypeMeta `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+    Spec FooMachinePoolSpec `json:"spec,omitempty"`
+    Status FooMachinePoolStatus `json:"status,omitempty"`
+}
+
+type FooMachinePoolSpec struct {
+    // See other rules for more details about mandatory/optional fields in InfraMachinePool spec.
+    // Other fields SHOULD be added based on the needs of your provider.
+}
+
+type FooMachinePoolStatus struct {
+    // See other rules for more details about mandatory/optional fields in InfraMachinePool status.
+    // Other fields SHOULD be added based on the needs of your provider.
+}
+```
+
+For each InfraMachinePool resource, you MUST also add the corresponding list resource.
+The list resource MUST be named as `<InfraMachine>List`.
+
+```go
+// +kubebuilder:object:root=true
+
+// FooMachinePoolList contains a list of foomachinepools.
+type FooMachinePoolList struct {
+    metav1.TypeMeta `json:",inline"`
+    metav1.ListMeta `json:"metadata,omitempty"`
+    Items           []FooMachinePool `json:"items"`
+}
+```
+
+### InfraMachinePool: infrastructureMachineKind
+
+If a providers implementation of a InfraMachinePool supports "MachinePool Machines" (where all the replicas in a MachinePool can be represented by a Machine & InfraMachine) then specifying and supplying a value for this field signals to Cluster API that the provider is opted-in to MachinePoolMachines.
+
+If you want to adopt MachinePool Machines then you MUST have a `status.infrastructureMachineKind` field and the field must contain the resource kind of the InfraMachine that represent the replicas of the pool. For example, for the AWS provider the value would be set to `AWSMachine`.
+
+```go
+type FooMachinePoolStatus struct {
+	// InfrastructureMachineKind is the kind of the infrastructure resources behind MachinePool Machines.
+	// +optional
+	InfrastructureMachineKind string `json:"infrastructureMachineKind,omitempty"`
+   
+    // See other rules for more details about mandatory/optional fields in InfraMachinePool status.
+    // Other fields SHOULD be added based on the needs of your provider.
+}
+```
+
+By opting into MachinePool Machines its the responsibility of the provider to create an instance of a InfraMachine for every replica and manage their lifecycle.
+
+### InfraMachinePool: instances
```

Comment:
I would suggest to add a definition to what an instance is in the context of MP

Also, if this field is not used by CAPI let's make it clear.
e.g.

Please note that this field is not used by CAPI. Nevertheless, it is documented in this contract to foster design choice that will ensure a consistent user experience across all the MachinePool implementations,